lib/rails/instrumentation/subscribers/action_controller_subscriber.rb

```diff
@@ -103,13 +103,22 @@ def process_action(event)
             'span.kind' => 'server'
           )
 
+
+          status_code = event.payload[:status]
+          if status_code.is_a? Integer and status_code >= 500
+            tags['error'] = true
+          end
+
           # Only append these tags onto the active span created by the patched 'process_action'
           # Otherwise, create a new span for this notification as usual
           active_span = ::Rails::Instrumentation.tracer.active_span
           if active_span && active_span.operation_name == span_name
             tags.each do |key, value|
               ::Rails::Instrumentation.tracer.active_span.set_tag(key, value)
             end
+            if event.payload.key? :exception
```

Comment:
this doesn't work with `rescue_from`, in which case we have to rely on the `status_code`